Lib/contextlib.py

```diff
@@ -143,6 +146,8 @@ def __enter__(self):
             raise RuntimeError("generator didn't yield") from None
 
     def __exit__(self, typ, value, traceback):
+        # don't keep the stored exception alive unnecessarily
+        self.exc_context, exc_context = None, self.exc_context
```

Comment:
FWIW, it's easy to miss what's going on here.  Outside the simplest cases, it's almost always better to split this form up:

```suggestion
        exc_context = self.exc_context
        self.exc_context = None
```